### PR TITLE
Embetter pofile sorting

### PR DESCRIPTION
**Pull request by akx**
I think sorting message locations irrespective of the sort flags is alright.

The only UC I can think of where that wouldn't be desired might be trying to read a .po and emitting an identical copy of it, but I don't think Babel does that as it is anyway?

### CODE REVIEW

**Comment by akx** [comment]
@jtwang DRYed! Good point.

File: babel/messages/pofile.py [hunk 1]
```diff
@@ -474,8 +472,21 @@ def _write_message(message, prefix=''):
...
+            sort_by_message=sort_output,
+            sort_by_file=sort_by_file
+        ):
             for comment in message.user_comments:
                 _write_comment(comment)
             _write_message(message, prefix='#~ ')
             _write('\n')
+
+
+def _sort_messages(messages, sort_by_message, sort_by_file):
```
Comment (jtwang) [inline babel/messages/pofile.py]: Test please and docs. No default values for sort flags?

**Comment by akx** [comment]
@jtwang: I improved the signature of the helper function a little more -- no good reason to have two mutually exclusive arguments!

Also, I don't think it warrants a separate test; that function's coverage should already be 100% by way of the other tests.

**Comment by jtwang** [comment]
ack 4f60b3ebde4a462ab686129982dc13f3d840617a, bc59c9c64ed285888a1bd929d55abbb63212e95b, edce8eeaa594e277d3abb29fb419cf6e4fefb52a, 65ce160b6915011e8c99e6175607e5fb329ce7fb

@akx totally disagree about testing, but I'll defer to your familiarity with this project. :)

Single arg change looks good (bah enums only in 3.4). Good to merge!

**Comment by akx** [comment]
Cheers @jtwang! :)

**Pull request merged**